python/ml4ir/base/features/preprocessing.py

```diff
@@ -46,12 +42,23 @@ def preprocess_text(feature_tensor, remove_punctuation=False, to_lower=False):
         feature_tensor: input feature tensor of type tf.string
         remove_punctuation: bool; whether to remove punctuation characters from strings
         to_lower: bool; whether to convert string to lower case
+        punctuation: punctuation characters to replace (a single string containing the character to remove
+        replace_with_whitespace: if True punctuation will be replaced by whitespace (i.e. used as separator), note that
+                               leading and trailing whitespace will also be removed, as well as consecutive whitespaces.
 
     Returns:
         processed float tensor
     """
     if remove_punctuation:
-        feature_tensor = tf.strings.regex_replace(feature_tensor, PUNCTUATION_REGEX, "")
+        replacement = ""
```

Comment:
we could make this more pythonic as `replacement = " " if replace_with_whitespace else ""`